### PR TITLE
chore(release): prepare 3.3.0 release

### PR DESCRIPTION
**Pull request by tao12345666333**
**What this PR does / why we need it**:

**Which issue this PR fixes**:

part of https://github.com/Kong/kubernetes-ingress-controller/issues/6439 

**Special notes for your reviewer**:

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [x] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -96,7 +97,9 @@ Adding a new version? You'll need three changes:
  - [0.0.5](#005)
  - [0.0.4 and prior](#004-and-prior)
 
-## Unreleased
+## [3.3.0]
+
+> Release date: 2024-08-26
```
Comment (randmonkey) [inline CHANGELOG.md:102]: Should it be 08-27?

**Review by randmonkey — COMMENTED** [review]
I am not sure if we want to add highlights and what should be the highlights in this PR.

File: CHANGELOG.md (hunk 1 above)
Comment (tao12345666333) [reply CHANGELOG.md:102]: The code was checked out on 08.26 and has not been modified. The current lack of merge is only due to some issues with the CI process. Let's keep it as it is on 08.26 and stick to our original plan.

In addition, if considering the time zone, there are still some places at 08.26. :rofl:

**Review by randmonkey — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (randmonkey) [reply CHANGELOG.md:102]: In Niue it is still 08.26 :-)

**Pull request merged**